site/scripts/prepare-blocks.ts

```diff
@@ -154,37 +148,42 @@ const ensureRepositorySnapshot = async ({
     return repositorySnapshotDirPath;
   }
 
-  const { path: unzipDirPath, cleanup: cleanupUnzipDirPath } = await tmp.dir({
+  const { path: untarDirPath, cleanup: cleanupUntarDirPath } = await tmp.dir({
     unsafeCleanup: true,
   });
 
   try {
-    console.log(chalk.green(`Downloading ${zipUrl}...`));
-    // @todo Consider finding cross-platform NPM packages for curl and unzip commands
+    console.log(chalk.green(`Downloading ${tarballUrl}...`));
+    // @todo Consider finding cross-platform NPM packages for curl and untar commands
     await execa(
       "curl",
-      ["-sL", "-o", path.resolve(unzipDirPath, `repo.zip`), zipUrl],
+      ["-sL", "-o", path.resolve(untarDirPath, `repo.tar.gz`), tarballUrl],
       defaultExecaOptions,
     );
 
+    const untarPath = path.resolve(untarDirPath, "repo");
+
+    console.log(chalk.green(`Creating output directory...`));
+    await execa("mkdir", ["-p", untarPath]);
+
     console.log(chalk.green(`Unpacking archive...`));
-    const unzipPath = path.resolve(unzipDirPath, "repo");
+
     await execa(
-      "unzip",
-      ["-q", path.resolve(unzipDirPath, "repo.zip"), "-d", unzipPath],
+      "tar",
+      ["-xvf", path.resolve(untarDirPath, "repo.tar.gz"), "-C", untarPath],
       defaultExecaOptions,
     );
 
-    const innerDir = await fs.readdir(unzipPath);
+    const innerDir = await fs.readdir(untarPath);
```

Comment:
```suggestion
    const innerDirName = (await fs.readdir(...))[0]!;
```

```suggestion
    const itemsInsideOutputDir = await fs.readdir(outputDirPath);
```

The original naming was not perfect – my bad. Because `fs.readdir` returns a list of items, it’d be nice to reflect that in variable name or extract the element we need right away. So plural for an array or `[0]!` on the same line.

I prefer to be explicit about what a fs-related variable contains. Hence suffixes like `xyzDirPath`, `abcFileName` etc. When suffixes are omitted, it’s easier to pass wrong variables around as args.